src/GafferSceneModule/CoreBinding.cpp

```diff
@@ -130,6 +132,43 @@ struct ScenePathFromString
 
 };
 
+// As a convenience we also accept lists in place of ScenePaths when
+// calling from python.
+struct ScenePathFromList
+{
+
+	ScenePathFromList()
+	{
+		boost::python::converter::registry::push_back(
+			&convertible,
+			&construct,
+			boost::python::type_id<ScenePlug::ScenePath>()
+		);
+	}
+
+	static void *convertible( PyObject *obj )
+	{
+		extract<boost::python::list> e( obj );
+		if( e.check() )
+		{
+			return obj;
+		}
+
+		return nullptr;
+	}
+
+	static void construct( PyObject *obj, boost::python::converter::rvalue_from_python_stage1_data *data )
+	{
+		void *storage = (( converter::rvalue_from_python_storage<ScenePlug::ScenePath>* ) data )->storage.bytes;
+		ScenePlug::ScenePath *path = new( storage ) ScenePlug::ScenePath();
+		data->convertible = storage;
+
+		boost::python::list l = extract<boost::python::list>( obj );
+		boost::python::container_utils::extend_container( *path, l );
```

Comment:
> it would take a bit of a dive through the Boost sources to determine if there was a leak here or not.

I dipped my toe in and was dragged down by a templated metashark. But I _think_ `storage` will get deleted appropriately if we throw, because I _think_ it is owned by `obj_mgr_arg_from_python`, which destroys it in the destructor via `python::detail::destroy_referent()`.

That coupled with my interactive testing gives me enough confidence to go ahead with the code as-is and with the additional ScenePlugTest checks.